### DISCUSSION TITLE
Segmentation fault while executing run_video_slam

### DISCUSSION THREAD

**Discussion by Shitikantha-Bagh** [stella-cv/stella_vslam#611] [Q&A]
Hello @ymd-stella , Completely respect your time. Please have a look into below error, and let me know if you need any more input.

I have been working on the marker integration, so I just tried to store the markers data into the msgpack, and changed related files. Then built the stella vslam again with -DCMAKE_BUILD_TYPE=RelWithDebInfo ..   make -j4  sudo make install.

I was getting the desired output, then I started to store reference keyframe and update local BA and global BA, but it was not showing good results, so commented out that part and built again. Now when I am executing the below command

`./run_video_slam -v ./orb_vocab.fbow -m ./video/VID_20240508_130714_00_042.mp4 -c ~/lib_updated/stella_vslam/example/aist/equirectangular_marker.yaml --frame-skip 10 --no-sleep --map-db-out mapexample.msg`

Here is the output I get:

[2024-07-02 11:22:43.846] [I] config file loaded: /home/sbagh/lib_updated/stella_vslam/example/aist/equirectangular_marker.yaml
--start-timestamp is not set. using system timestamp.
If --no-sleep is set without --start-timestamp, timestamps may overlap between multiple runs.
[2024-07-02 11:22:43.846] [I]
original version of OpenVSLAM,
Copyright (C) 2019,
National Institute of Advanced Industrial Science and Technology (AIST)
All rights reserved.
stella_vslam (the changes after forking from OpenVSLAM),
Copyright (C) 2022, stella-cv, All rights reserved.

This is free software,
and you are welcome to redistribute it under certain conditions.
See the LICENSE file.

Camera:
  name: Insta 360 X3
  setup: monocular
  model: equirectangular
  fps: 30.0
  cols: 3840
  rows: 1920
  color_order: RGB
Preprocessing:
  min_size: 1920
  mask_rectangles:
    - [0.0, 1.0, 0.0, 0.1]
    - [0.0, 1.0, 0.84, 1.0]
    - [0.0, 0.2, 0.7, 1.0]
    - [0.8, 1.0, 0.7, 1.0]
Feature:
  name: default ORB feature extraction setting
  scale_factor: 1.2
  num_levels: 8
  ini_fast_threshold: 20
  min_fast_threshold: 7
Mapping:
  backend: g2o
  baseline_dist_thr_ratio: 0.03
  redundant_obs_ratio_thr: 0.95
  num_covisibilities_for_landmark_generation: 20
  num_covisibilities_for_landmark_fusion: 20
  residual_deg_thr: 0.4
Tracking:
  backend: g2o
LoopDetector:
  backend: g2o
  enabled: true
  reject_by_graph_distance: true
  min_distance_on_graph: 50
GraphOptimizer:
  min_num_shared_lms: 200
GlobalOptimizer:
  thr_neighbor_keyframes: 100
System:
  map_format: msgpack
  num_grid_cols: 96
  num_grid_rows: 48
MarkerModel:
  type: aruco
  width: 0.3
  marker_size: 4
  max_markers: 100

[2024-07-02 11:22:43.847] [I] loading ORB vocabulary: ./orb_vocab.fbow
[2024-07-02 11:22:43.865] [I] load orb_params "default ORB feature extraction setting"
[2024-07-02 11:22:43.866] [I] startup SLAM system
[2024-07-02 11:22:43.866] [I] start global optimization module
[2024-07-02 11:22:43.866] [I] start mapping module
Stack trace (most recent call last) in thread 8544:
#10   Object "", at 0xffffffffffffffff, in
BFD: DWARF error: section .debug_info is larger than its filesize! (0x93f189 vs 0x531098)
#9    Object "/lib/x86_64-linux-gnu/libc.so.6", at 0x7fd5c37b6352, in clone
#8    Source "/build/glibc-LcI20x/glibc-2.31/nptl/pthread_create.c", line 477, in start_thread [0x7fd5c410e608]
#7    Object "/lib/x86_64-linux-gnu/libstdc++.so.6", at 0x7fd5c397adf3, in std::error_code::default_error_condition() const
#6  | Source "/usr/include/c++/9/thread", line 195, in _M_run
    |   194: void
    | > 195: _M_run() { _M_func(); }
    |   196:       };
    | Source "/usr/include/c++/9/thread", line 251, in operator()
    |   249:  using _Indices
    |   250:    = typename _Build_index_tuple<tuple_size<_Tuple>::value>::__type;
    | > 251:  return _M_invoke(_Indices());
    |   252: }
    |   253:       };
    | Source "/usr/include/c++/9/thread", line 244, in _M_invoke<0>
    |   242:  typename __result<_Tuple>::type
    |   243:  _M_invoke(_Index_tuple<_Ind...>)
    | > 244:  { return std::__invoke(std::get<_Ind>(std::move(_M_t))...); }
    |   245:
    |   246: typename __result<_Tuple>::type
    | Source "/usr/include/c++/9/bits/invoke.h", line 95, in __invoke<mono_tracking(const std::shared_ptr<stella_vslam::system>&, const std::shared_ptr<stella_vslam::config>&, const string&, const string&, unsigned int, unsigned int, bool, bool, bool, const string&, const string&, double, const string&)::<lambda()> >
    |    93:       using __type = typename __result::type;
    |    94:       using __tag = typename __result::__invoke_type;
    | >  95:       return std::__invoke_impl<__type>(__tag{}, std::forward<_Callable>(__fn),
    |    96: std::forward<_Args>(__args)...);
    |    97:     }
    | Source "/usr/include/c++/9/bits/invoke.h", line 60, in __invoke_impl<void, mono_tracking(const std::shared_ptr<stella_vslam::system>&, const std::shared_ptr<stella_vslam::config>&, const string&, const string&, unsigned int, unsigned int, bool, bool, bool, const string&, const string&, double, const string&)::<lambda()> >
    |    58:     constexpr _Res
    |    59:     __invoke_impl(__invoke_other, _Fn&& __f, _Args&&... __args)
    | >  60:     { return std::forward<_Fn>(__f)(std::forward<_Args>(__args)...); }
    |    61:
    |    62:   template<typename _Res, typename _MemFun, typename _Tp, typename... _Args>
      Source "/home/sbagh/lib_updated/stella_vslam_examples/src/run_video_slam.cc", line 166, in std::thread::_State_impl<std::thread::_Invoker<std::tuple<mono_tracking(std::shared_ptr<stella_vslam::system> const&, std::shared_ptr<stella_vslam::config> const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned int, unsigned int, bool, bool, bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, double, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)::{lambda()#1}> > >::_M_run() [0x55d61c252f8b]
        164:             if (!frame.empty() && (num_frame % frame_skip == 0)) {
        165:                 // input the current frame and estimate the camera pose
      > 166:                 slam->feed_monocular_frame(frame, timestamp, mask);
        167:             }
        168:
        169:             const auto tp_2 = std::chrono::steady_clock::now();
Segmentation fault (core dumped)

I found a similiar closed issue #455 and followed the steps to install Fbow again, but got same error. Please let me know how to rectify and what is the cause of error. Thank you.

**Comment by Shitikantha-Bagh** [comment]
Can anyone  please help me with this ?

**Reply by Skwangles** [reply]
Edit: Apologies, I have reread your intro, my steps are not very helpful, so I've rewritten my response. 

Does the system work when you uncomment the local BA/global BA parts? The error saying .debug-info size is larger than filesize does sound like a library/build issue related to RelWithDebug from a layman's perspective. Does building with just Release work?

**Reply by Shitikantha-Bagh** [reply]
Thanks for responding. I hope you understood my issue. Optimization codes work perfectly, I just wanted to explain the situation before   this error occured. I was working on marker integration, so wrote some codes for the markers optimization, as it didnt work out, I just commented that part and not the entire optimization part. And after successful compilation again, when I run SLAM, I get the above error. Also, I have reproduced the same errors by perfoming similiar activity after a new installation. Therefore, it is important to figure out whether this error would always appear .

I feel some variable, pointers etc. needs  to be reset. I found a similiar closed issue https://github.com/stella-cv/stella_vslam/issues/455.
